nemo_aligner/servers/server_callables.py

```diff
@@ -11,96 +11,47 @@
-    if sequence_lengths is not None:
-        assert len(inputs) == len(sequence_lengths)
-        assert extra_sequence_lengths == extra
-
-    try:
-        *list_outputs, exceeded = infer_fn(inputs=inputs, sequence_length=sequence_lengths, add_EOS=add_EOS)
-
-        processed_outputs = []
-
-        for output in list_outputs:
-            output = torch.cat(output, dim=0)
-            # unpad
-            output = output[: output.size(0) - extra]
-
-            processed_outputs.append(output.cpu().numpy())
-
-        exceeded = exceeded[: len(exceeded) - extra]
-
-    except RuntimeError as e:
-        raise PyTritonUnrecoverableError(f"Fatal error occurred - no further inferences possible. {e}") from e
-
-    return (*processed_outputs, np.array(exceeded, dtype=np.int32).reshape(-1, 1))
-
-
-class RewardModelCallable:
-    def __init__(self, *, model_name: str, infer_fn: callable, lock: threading.Lock):
-        self.model_name = model_name
-        self.lock = lock
-        self.infer_fn = infer_fn
-        self.inputs = (
-            Tensor(name="sentences", shape=(-1,), dtype=bytes, optional=True),
-            Tensor(name="tokens", shape=(-1,), dtype=np.int64, optional=True),
-            Tensor(name="sequence_lengths", shape=(-1,), dtype=np.int64, optional=True),
-            Tensor(name="add_EOS", shape=(1,), dtype=np.bool_, optional=True),
-        )
-        self.outputs = (
-            Tensor(name="rewards", shape=(1,), dtype=np.float32),
-            Tensor(name="exceeded", shape=(1,), dtype=np.int32),
+        tokens, sequence_lengths = tokenize_func(sentences)
+        sequence_lengths = sequence_lengths.unsqueeze(-1)
+    else:
+        tokens = torch.as_tensor(inputs["tokens"], dtype=torch.long, device=torch.cuda.current_device())
+        sequence_lengths = torch.as_tensor(
+            inputs["sequence_lengths"], dtype=torch.long, device=torch.cuda.current_device()
         )
 
-    @batch
-    @lock_method("self.lock")
-    def infer(self, **inputs: np.ndarray) -> Dict[str, np.ndarray]:
-        choice = ServerSignal.FORWARD.cuda()
-        torch.distributed.broadcast(choice, 0)
-
-        rewards, exceeded = run_rm_or_critic_inference(self.infer_fn, inputs=inputs)
-
-        output_dict = {
-            "rewards": rewards,
-            "exceeded": exceeded,
-        }
-        return output_dict
+    # strip along sequence dim
+    prestrip_sequence_length = tokens.shape[1]
+    if strip_sequence_length_to_multiple is not None:
+        stripped_sequence_length = (
+            math.ceil(sequence_lengths.max().item() / strip_sequence_length_to_multiple)
+            * strip_sequence_length_to_multiple
+        )
+        if stripped_sequence_length < prestrip_sequence_length:
+            tokens = tokens[:, :stripped_sequence_length]
+
+    # padding on the batch dim
+    _, amount_to_pad = divmod(tokens.size(0), pad_to_multiple)
+    if amount_to_pad > 0:
+        amount_to_pad = pad_to_multiple - amount_to_pad
```

Comment:
This is correct but suggestion for readability (at least for me!):
```suggestion
    num_extra = tokens.size(0) % pad_to_multiple
    amount_to_pad = 0 if num_extra == 0 else pad_to_multiple - num_extra
```